kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/KafkaConnectService.java

```diff
@@ -0,0 +1,227 @@
+                                        withRetryOnConflict(KafkaConnectClients.withBaseUrl(connect).getConnectorTaskStatus(connectorName, task.getId().getTask()))
+                                                .map(kafkaConnectMapper::fromClient)
+                                                .map(task::status)
+                                )
+                );
+    }
+
+    public Mono<Void> restartConnectorTask(String clusterName, String connectName, String connectorName, Integer taskId) {
+        return getConnectAddress(clusterName, connectName)
+                .flatMap(connect ->
+                        withRetryOnConflict(
+                                KafkaConnectClients.withBaseUrl(connect).restartConnectorTask(connectorName, taskId)
+                        )
+                );
+    }
+
+    public Mono<Flux<ConnectorPlugin>> getConnectorPlugins(String clusterName, String connectName) {
+        return Mono.just(getConnectAddress(clusterName, connectName)
+                .flatMapMany(connect ->
+                        withRetryOnConflict(
+                                KafkaConnectClients.withBaseUrl(connect).getConnectorPlugins()
+                        )
+                                .map(kafkaConnectMapper::fromClient)
+                ));
+    }
+
+    public Mono<ConnectorPluginConfigValidationResponse> validateConnectorPluginConfig(String clusterName, String connectName, String pluginName, Mono<Object> requestBody) {
+        return getConnectAddress(clusterName, connectName)
+                .flatMap(connect ->
+                        requestBody.flatMap(body ->
+                                withRetryOnConflict(
+                                        KafkaConnectClients.withBaseUrl(connect).validateConnectorPluginConfig(pluginName, (Map<String, Object>) body)
+                                ))
+                                .map(kafkaConnectMapper::fromClient)
+                );
+    }
+
+    private Mono<KafkaCluster> getCluster(String clusterName) {
+        return clustersStorage.getClusterByName(clusterName)
+                .map(Mono::just)
+                .orElse(Mono.error(new NotFoundException("No such cluster")));
+    }
+
+    private Mono<String> getConnectAddress(String clusterName, String connectName) {
+        return getCluster(clusterName)
+                .map(kafkaCluster ->
+                        kafkaCluster.getKafkaConnect().stream()
+                                .filter(connect -> connect.getName().equals(connectName))
+                                .findFirst()
+                                .map(KafkaConnectCluster::getAddress)
+                )
+                .flatMap(connect -> connect
+                        .map(Mono::just)
+                        .orElse(Mono.error(new NotFoundException("No such connect cluster")))
+                );
+    }
+
+    private <T> Mono<T> withRetryOnConflict(Mono<T> publisher) {
```

Comment:
you call this method on every call with KafkaConnectClients, maybe it make sense to do it there?

Reply:
Fixed